src/main/java/com/softserve/edu/sporthubujp/controller/UserController.java

```diff
@@ -0,0 +1,40 @@
+package com.softserve.edu.sporthubujp.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.softserve.edu.sporthubujp.dto.UserDTO;
+import com.softserve.edu.sporthubujp.dto.UserSaveProfileDTO;
+import com.softserve.edu.sporthubujp.service.UserService;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@RestController
+@RequestMapping("/api/v1")
+public class UserController {
+
+    private final UserService userService;
+
+    @Autowired
+    public UserController(UserService userService) {
+        this.userService = userService;
+    }
+
+    @PutMapping(path = "/profile/edit/{id}")
+    @PreAuthorize("hasAnyAuthority('ADMIN', 'USER')")
+    public ResponseEntity<UserDTO> updateProfile(@RequestBody UserSaveProfileDTO newUser,
+        @PathVariable("id") String id) {
```

Comment:
Do not pass id, you need to use Principal, because your user can update any other user info :)